frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices-mode/install.tsx

```diff
@@ -32,31 +41,52 @@ export const CreateAttachedDevicesCluster: React.FC<CreateAttachedDevicesCluster
   const [isLsoPresent, setIsLsoPresent] = React.useState(false);
   const [allDataLoaded, setAllDataLoaded] = React.useState(false);
   const [lsoNs, setLsoNs] = React.useState('');
+  const [lsoVersion, setLsoVersion] = React.useState('');
+
+  const subscriptionResource: WatchK8sResource = {
+    kind: referenceForModel(SubscriptionModel),
+    isList: true,
+  };
   const [subscription, subscriptionLoaded, subscriptionLoadError] = useK8sWatchResource<
     K8sResourceKind[]
-  >(LSOSubscriptionResource);
+  >(subscriptionResource);
+
+  const csvResource = {
+    kind: referenceForModel(ClusterServiceVersionModel),
+    fieldSelector: `metadata.name=${lsoVersion}`,
+    namespace: lsoNs,
+    isList: true,
+  };
+  const [csv, csvLoaded, csvLoadError] = useK8sWatchResource<K8sResourceKind[]>(csvResource);
 
   React.useEffect(() => {
-    if (subscriptionLoadError || (!subscription.length && subscriptionLoaded)) {
+    const lsoSubscription = subscription.filter((item) => item?.spec?.name === LSO_OPERATOR);
+    setLsoNs(getNamespace(lsoSubscription[0]));
+    setLsoVersion(getOperatorVersion(lsoSubscription[0]));
```

Comment:
Since its just one element only hence use `find`.
```suggestion
    const lsoSubscription = subscription.find((item) => item?.spec?.name === LSO_OPERATOR);
    setLsoNs(getNamespace(lsoSubscription));
    setLsoVersion(getOperatorVersion(lsoSubscription));
```